synthesizer/src/process/stack/inclusion/mod.rs

```diff
@@ -671,13 +727,178 @@ mod tests {
+            // Prepare the assignments.
+            let (assignments, global_state_root) = {
+                let execution = cast_ref!(execution as Execution<CurrentNetwork>);
+                let inclusion = cast_ref!(inclusion as Inclusion<CurrentNetwork>);
+                inclusion.prepare_execution(execution, query)?
+            };
+            let assignments = cast_ref!(assignments as Vec<InclusionAssignment<CurrentNetwork>>);
+            let global_state_root = *cast_ref!((*global_state_root) as Field<CurrentNetwork>);
+
+            // Compute the inclusion proof and update the execution.
+            let execution = inclusion.prove_execution::<CurrentAleo, _>(
+                execution,
+                assignments,
+                global_state_root.into(),
+                Some(inclusion_proving_key),
+                rng,
+            )?;
+
+            // Prepare the return.
+            let response = cast_ref!(response as Response<CurrentNetwork>).clone();
+            let execution = cast_ref!(execution as Execution<CurrentNetwork>).clone();
+            let metrics = cast_ref!(metrics as Vec<CallMetrics<CurrentNetwork>>).clone();
+
+            // Return the response, execution, and metrics.
+            Ok((response, execution, inclusion, metrics))
+        }
+
+        //----------- FUNCTIONS ASSUMED TO BE EXECUTED OUTSIDE OF WEBASSEMBLY -----------
+        // This function would be non-wasm function (most likely in javascript that takes the
+        // "commitments" output from get_authorization_offline and use it to make a call in
+        // the outside environment to the Aleo API to get the state roots and state paths
+        fn get_inclusion_proof_data(_commitments: &[Field<CurrentNetwork>]) -> (&str, &str) {
+            (STATE_ROOT, COMMITMENT_MAP)
+        }
+
+        //----------- SAMPLE WEB PROVING PROGRAM FLOW -----------
+        // This program execution flow would be executed from javascript
+
+        // ENTER WASM: Get authorization and commitments
+        let (authorization, commitments) =
+            authorize_offline(&caller_private_key, "credits.aleo", "transfer", [r0, r1, r2].into_iter()).unwrap();
+
+        // LEAVE WASM: Get this data from the web (not in rust)
+        let (state_root, commitment_map) = get_inclusion_proof_data(&commitments); // This would be inserted into the main
+
+        // ENTER WASM: Execute the program in wasm with data supplied from an outside environment
+        let (_, execution, _, _) = execute_program(
+            "credits.aleo",
+            "transfer",
+            authorization,
+            state_root,
+            commitment_map,
+            credits_proving_key,
+            inclusion_proving_key,
+        )
+        .unwrap();
+
+        // ENTER WASM: Verify the execution
+        assert!(Inclusion::verify_execution(&execution, Some(inclusion_verifying_key)).is_ok());
+    }
```

Comment:
This is a sample test that mocks how it would work in a webassembly environment where namely:
* The program proving keys, inclusion keys are inserted into the proving process
* In a webassembly context, the data needed to make the inclusion proof is called from javascript and inserted back into another function.